Support Python 3.9

typing_inspect doesn't really work with Python 3.9 right now. There are issues like #60 and #64, but a variety of things need to be updated.
My initial survey (just guessing from poking at implementations; it would be better to have someone review/expand this list who is more intimately familiar with the code-level changes):
* The typing module in 3.9 introduces a new "_BaseGenericAlias" class that might more correctly correspond to typing_inspect's current uses of "_GenericAlias."
* Add support and tests for the new plain type hints (e.g. "list[int]" rather than "List[int]"). There seems to be a new "GenericAlias" class (that extends no other classes) for these kind of annotations.
* Get the current test suite to pass under 3.9.
* Add 3.9 to the travis config for regression testing.

Any volunteers willing to give it a try? @ilevkivskyi - any additional guidance? What's wrong/missing in my above list?

Adding this bit of info here, in case it helps: currently, there's only one test suite failure:

```
======================================================================
FAIL: test_generic (test_typing_inspect.IsUtilityTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/guix-build-python-typing-inspect-0.6.0.drv-0/typing_inspect-0.6.0/test_typing_inspect.py", line 133, in test_generic
    self.sample_test(is_generic_type, samples, nonsamples)
  File "/tmp/guix-build-python-typing-inspect-0.6.0.drv-0/typing_inspect-0.6.0/test_typing_inspect.py", line 123, in sample_test
    self.assertTrue(fun(s), msg=msg % (fun.__name__, str(s), 'True'))
AssertionError: False is not true : Error asserting that is_generic_type(typing.Mapping) is True

----------------------------------------------------------------------
```

--- a/typing_inspect.py
+++ b/typing_inspect.py
@@ -11,6 +11,9 @@ import sys
 from mypy_extensions import _TypedDictMeta as _TypedDictMeta_Mypy
 if sys.version_info[:3] >= (3, 4, 0) and sys.version_info[:3] < (3, 9, 0):
     from typing_extensions import _TypedDictMeta as _TypedDictMeta_TE
+elif sys.version_info[:3] >= (3, 9, 0):
+    # typing_extensions.TypedDict is a re-export from typing.
+    from typing import _TypedDictMeta as _TypedDictMeta_TE
 else:
     # typing_extensions.TypedDict is a re-export from typing.
     from typing import TypedDict

